runtime/ops/fs.rs

```diff
@@ -501,6 +501,7 @@ async fn op_funlock_async(
 }
 
 #[op]
+#[allow(clippy::unnecessary_cast)]
```

Comment:
Thanks for the link! Then I wonder if it's not enough for us to just remove `as u32`, just like the clippy suggested, rather than suppressing this lint. Do you have any idea?

Reply:
@magurotuna I try to remove `#[allow(clippy::unnecessary_cast)]` and `as u32` in this [action job](https://github.com/linbingquan/deno/actions/runs/3713679510/jobs/6296617324), I found it was passed lint test script, but it also failed `fastci` in my repo...

I think I try to remove those code in this PR, maybe it will pass all test.

Reply:
Unfortunately, removed `#[allow(clippy::unnecessary_cast)]` and `as u32` code does not pass the test.

https://github.com/denoland/deno/actions/runs/3713884681/jobs/6297078283#step:24:507

```
error[E0308]: mismatched types
   --> runtime/ops/fs.rs:528:8
    |
528 |     Ok(r.bits())
    |     -- ^^^^^^^^ expected `u32`, found `u16`
    |     |
    |     arguments to this enum variant are incorrect
    |
note: tuple variant defined here
help: you can convert a `u16` to a `u32`
    |
528 |     Ok(r.bits().into())
    |                +++++++
```

I was add `into()` in my reop before this PR, it does not work... 
see: https://github.com/linbingquan/deno/actions/runs/3711219488/jobs/6291415984#step:22:502

Reply:
OK that's interesting. It seems like on macOS `r.bits()` returns `u16`, whereas on Linux it returns `u32`, which ends up us getting the `unnecessary_cast` warning for Linux, since cast from `u32` to `u32` is obviously unnecessary.
So I would suggest adding another `cfg` to make clippy happy, like following:

```rs
    #[cfg(target_os = "linux")]
    {
      Ok(r.bits())
    }
    #[cfg(target_os = "macos")]
    {
      Ok(r.bits() as u32)
    }
```

I believe this works and we can remove `#[allow(clippy::unnecessary_cast)]` : )

Reply:
> It seems like on macOS `r.bits()` returns `u16`, whereas on Linux it returns `u32`.

It makes scene.

I removed `#[allow(clippy::unnecessary_cast)]` and added your suggestion code in this [commit](https://github.com/denoland/deno/pull/17078/commits/ceed5b3b209d97536b16bcd25bb0c29adc3e26ba),
it was passed lint test, but it has an [error](https://github.com/denoland/deno/actions/runs/3718668155/jobs/6307020794#step:33:3648) failed at `fastci windows-2019`.
En... I don't know what caused this error.

Reply:
It's most likely because of the flakiness of that test case. So I believe if you rerun the CI it should be green.
https://github.com/denoland/deno/issues/16041
Can you rerun CI again? (you can do so by doing an empty commit, if needed) Let's try to see if it succeeds.

Reply:
Thank you for your help. It was succeeded to pass the test. 
see: https://github.com/denoland/deno/actions/runs/3719563478